airbyte-integrations/connectors/source-datadog/source_datadog/streams.py

```diff
@@ -286,3 +304,112 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
+        self._name = value
+
+    @property
+    def cursor_field(self) -> Union[str, List[str]]:
+        return "sync_date"
+
+    def request_body_json(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Optional[Mapping]:
+
+        if self.query_end_date:
+            end_date = int(parse(self.query_end_date).timestamp() * 1000)
+        else:
+            end_date = int(datetime.now().timestamp()) * 1000
+
+        if self.query_start_date:
+            start_date = int(parse(self.query_start_date).timestamp() * 1000)
+        elif self._cursor_value:
+            start_date = int(parse(self._cursor_value).timestamp() * 1000)
+        else:
+            start_date = int((datetime.now() - timedelta(hours=24)).timestamp()) * 1000
+
+        if self.data_source in ["metrics", "cloud_cost"]:
+            payload = {
+                "data": {
+                    "type": "timeseries_request",
+                    "attributes": {
+                        "to": end_date,
+                        "from": start_date,
+                        "queries": [
+                            {
+                                "data_source": self.data_source,
+                                "query": self.query_string,
+                                "name": self.name,
+                            }
+                        ],
+                    },
+                }
+            }
+        elif self.data_source in ["logs", "rum"]:
+            payload = {
+                "data": {
+                    "type": "timeseries_request",
+                    "attributes": {
+                        "to": end_date,
+                        "from": start_date,
+                        "queries": [
+                            {
+                                "data_source": self.data_source,
+                                "name": self.name,
+                                "search": {
+                                    "query": self.query_string
+                                },
+                                "compute": {
+                                    "aggregation": "count"
+                                },
+                            }
+                        ],
+                    },
+                }
+            }
+        return payload
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        local_json_schema = {
+            "$schema": "http://json-schema.org/draft-07/schema#",
+            "type": "object",
+            "properties": {},
+            "additionalProperties": True,
+        }
+        return local_json_schema
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        return [response.json()]
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        self._cursor_value = self.end_date
```

Comment:
Can you explain the logic behind this? Why aren't you getting the cursor from datadog response?

Reply:
The [datadog time series API](https://docs.datadoghq.com/api/latest/metrics/?code-lang=curl#query-timeseries-data-across-multiple-products) doesn’t return a cursor unlike other Datadog query response.